packages/@aws-cdk/cfn-include/lib/cdk-include.ts

```diff
@@ -0,0 +1,221 @@
+import * as core from '@aws-cdk/core';
+import * as fs from 'fs-extra';
+import * as path from 'path';
+
+/**
+ * Construction properties of {@link CdkInclude}.
+ */
+export interface CdkIncludeProps {
+  /**
+   * Path to the template file.
+   *
+   * Currently, only JSON templates are supported.
+   */
+  readonly templateFile: string;
+
+  /**
+   * Whether the resources should have the same logical IDs in the resulting CDK template
+   * as they did in the original CloudFormation template file.
+   *
+   * @default true
+   */
+  readonly preserveLogicalIds?: boolean;
+}
+
+/**
+ * A construct that allows you to include an existing CloudFormation template file into your CDK application.
+ * You can retrieve any of the resources defined in the template file by calling the {@link getResource} method.
+ * Any modifications you make on the returned resource objects will be reflected in the resulting CDK template.
+ *
+ * @see getResource
+ */
+export class CdkInclude extends core.Construct {
+  private readonly resources: { [logicalId: string]: core.CfnResource } = {};
+  private readonly cfnType2L1Class: { [cfnType: string]: string };
+  private readonly template: any;
+  private readonly preserveLogicalIds: boolean;
+
+  constructor(scope: core.Construct, id: string, props: CdkIncludeProps) {
+    super(scope, id);
+
+    // read the template into a JS object
+    this.template = fs.readJsonSync(props.templateFile);
+
+    // check for any sections of the template we don't recognize
+    for (const section of Object.keys(this.template)) {
+      switch (section) {
+        case 'AWSTemplateFormatVersion':
+        case 'Description':
+        case 'Resources':
+          // currently, we only support these 3 sections
+          break;
+        default:
```

Comment:
Shouldn't we at least pass-through sections that we don't recognize? Otherwise this is quite unusable for anything that's slightly complex, no?

Reply:
I think `AWSTemplateFormatVersion` and `Description` are the only 2 sections you can safely skip without affecting the semantics of the template - if you want to add more, let me know which ones.

Reply:
Not sure I understand. I am proposing that any entity that you don’t recognize you’ll just pass through without allowing users to interact with it. This way, this class can be a replacement of CfnInclude even before there is explicit support to all types of template elements.

Reply:
Just to clarify, to make sure I understand:

You want to override `_toCloudFormation()` in `CdkInclude` to render all sections of the input template unchanged, except the `Resources` section, which will be taken care of by the created L1s?

Reply:
That's one way to implement it (you can also create a "pass through" construct for each one and not at the `CdkInclude` level). The idea is that this will have feature parity with the current `CfnInclude`

Reply:
This will add a ton of complexity to this already large PR (and it's about to become quite a bit larger, with the code generation changes that you asked for).

We can't simply pass through sections like Parameters, Conditions and Outputs, because `preserveLogicalIds` could be `false`. We not only have to rewrite the logical ID of the CFN element, but also all references to it in expressions like `Ref`. We don't even support Conditions currently for Resources, so that would have to be added as well.

I'm fine with not deprecating `CfnInclude` in this PR, but waiting to do it until later, when `CdkInclude` has feature parity with `CfnInclude`.

Thoughts on this?

Reply:
Alternatively, we can ditch `preserveLogicalIds` for now, and assume it's always `true`, as it's a source of complexity not only here, but also in the L1 code generation changes that you asked for.

Let me know what you think.

Reply:
Sure, no worries about not implementing all of this right now (naturally we can't deprecate CfnInclude until this can actually replace it). I think the important part is the conceptual discussion about the design of this feature, not what does into this PR.

Perhaps the right way to work on this is to actually implement it in a separate repo until it's ready to be merged in. What is the value you see from releasing this with the CDK before it's ready?

Reply:
Decide to forego `preserveLogicalIds` equal to `false` for now, and implemented a simple pass through of all sections other than Resources.